scylla/src/transport/session.rs

```diff
@@ -180,22 +183,29 @@ impl Session {
     /// # Arguments
     ///#
     pub async fn prepare(&self, query: &str) -> Result<PreparedStatement, TransportError> {
-        // FIXME: Prepared statement ids are local to a node, so we must make sure
-        // that prepare() sends to all nodes and keeps all ids.
-        let result = self.any_connection()?.prepare(query.to_owned()).await?;
-        match result {
-            Response::Error(err) => Err(err.into()),
-            Response::Result(result::Result::Prepared(p)) => Ok(PreparedStatement::new(
-                p.id,
-                p.prepared_metadata,
-                query.to_owned(),
-            )),
-            _ => Err(TransportError::PrepareError(
-                PrepareError::InternalDriverError(InternalDriverError::UnexpectedResponse),
-            )),
+        let connections = self.get_connections()?;
+        let all_connections = connections.iter().flat_map(|(_, v)| v);
+
+        // Prepare statements on all connections concurrently
+        let handles = all_connections.map(|c| c.prepare(query));
+        let mut results = try_join_all(handles).await?;
+
+        if !Self::validate_prepared_ids_equality(&results) {
+            return Err(TransportError::PreparedStatementsIDsNotMatch);
+        }
+
+        match results.pop() {
+            Some(result) => Ok(result),
+            None => Err(TransportError::NoConnectionsAvailable),
         }
```

Comment:
Convert it to:
```rust
results.pop().ok_or_else(TransportError::NoConnectionsAvailable)
```

Reply:
Done (I chose to use `ok_or` instead of `ok_or_else`)